gems/sorbet-runtime/test/types/edge_cases.rb

```diff
@@ -297,4 +297,17 @@ def foo; end
     assert_match(/A previous invocation of #<UnboundMethod: /, e.message)
   end
 
+  it 'does not crash when two threads call the same wrapper' do
+    5000.times do
```

Comment:
could locks or some other control primitive be used to reproduce the error reliably?

Reply:
I can't think of a good way to do it since it requires very fine grained scheduling of the two threads inside sorbet-runtime's code. I suppose it's possible with some monkey patching but I think at that point it's not worth it.

This test takes one second to run right now, which is okay to me. I can remove it if it's a concern.

Reply:
I came up with 90ae979 which uses a barrier + a mutex + monkey patching to consistently reproduce the race. Not problem free of course, since this is now white box testing. If the wrapper code stops calling `run_sig_block_for_method` one day the test will lose value, and this also adds a development dependency on concurrent-ruby.

Let me know what you think.